src/azure-cli/azure/cli/command_modules/acr/manifest.py

```diff
@@ -0,0 +1,453 @@
+                                    password,
+                                    artifact_type=None):
+
+    result_list = {'references': []}
+    execute_next_http_call = True
+
+    params = {
+        'artifactType': artifact_type
+    }
+
+    while execute_next_http_call:
+        execute_next_http_call = False
+
+        result, next_link = request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=path,
+            username=username,
+            password=password,
+            result_index=None,
+            params=params)
+
+        if result:
+            result_list['references'].extend(result['references'])
+
+        if next_link:
+            # The registry is telling us there's more items in the list,
+            # and another call is needed. The link header looks something
+            # like `Link: </v2/_catalog?last=hello-world&n=1>; rel="next"`
+            # we should follow the next path indicated in the link header
+            next_link_path = next_link[(next_link.index('<') + 1):next_link.index('>')]
+            tokens = next_link_path.split('?', 1)
+            params = {y[0]: unquote(y[1]) for y in (x.split('=', 1) for x in tokens[1].split('&'))}
+
+            execute_next_http_call = True
+
+    return result_list
+
+
+def _parse_fqdn(cmd, fqdn, is_manifest=True):
+    try:
+        fqdn = fqdn.lstrip('https://')
+        reg_addr = fqdn.split('/', 1)[0]
+        registry_name = reg_addr.split('.', 1)[0]
+        reg_suffix = '.' + reg_addr.split('.', 1)[1]
+        manifest_id = fqdn.split('/', 1)[1]
+
+        _validate_login_server_suffix(cmd, reg_suffix)
+
+        # We must check for this here as the default tag 'latest' gets added in _parse_image_name
+        if not is_manifest and ':' in manifest_id:
+            raise InvalidArgumentValueError("The positional parameter 'ID' should not include a tag or digest.")
+
+        repository, tag, manifest = _parse_image_name(manifest_id, allow_digest=True)
+
+    except IndexError as e:
+        if is_manifest:
+            raise InvalidArgumentValueError(BAD_MANIFEST_FQDN) from e
+
+        raise InvalidArgumentValueError(BAD_REPO_FQDN) from e
+
+    return registry_name, repository, tag, manifest
+
+
+def _validate_login_server_suffix(cmd, reg_suffix):
+    cli_ctx = cmd.cli_ctx
+    login_server_suffix = get_login_server_suffix(cli_ctx)
+
+    if reg_suffix != login_server_suffix:
+        raise InvalidArgumentValueError(f'Provided registry suffix \'{reg_suffix}\' does not match the configured az'
+                                        f' cli acr login server suffix \'{login_server_suffix}\'. Check the'
+                                        ' \'acrLoginServerEndpoint\' value when running \'az cloud show\'.')
+
+
+def acr_manifest_list(cmd,
+                      registry_name=None,
+                      repository=None,
+                      ID=None,
```

Comment:
We do not recommend that parameter and variable names be in uppercase, and lowercase is more in line with the naming convention

Reply:
fixed